libraries/AP_AHRS/AP_AHRS.h

```diff
@@ -527,6 +530,12 @@ class AP_AHRS
         return false;
     }
 
+    // get a source's velocity innovations.  source should be from 0 to 7 (see AP_NavEKF_Source::SourceXY)
+    // returns true on success and results are placed in innovations and variances arguments
+    virtual bool get_vel_innovations_and_variances_by_source(uint8_t source, Vector3f &innovations, Vector3f &variances) const {
```

Comment:
I considered this but my understanding is that we need to limit the number of methods exposed to scripting because each takes up more memory or flash or something like that.

My scripts only make use of the innovations so I could drop the ability to retrieve variances if @priseborough thinks they won't be useful for determining the health of a source.

Reply:
Paul says there's no quick answer to this question of whether variances will be useful.. I'm not sure but I think we should just leave it like this until we have more examples of source switching scripts and understand better what they need to make good decisions.